tests/test_inference_types_serialization.py

```diff
@@ -6,17 +6,20 @@
     FullNormal,
     Mixture,
     Normal,
+    Point2Point2Range,
     Pose2AprilTag4Corners,
+    Pose2Point2BearingRange,
     Pose2Pose2,
     Prior,
+    PriorPoint2,
     PriorPose2,
     Uniform,
 )
 from src.navability.entities.factor.inferencetypes import LinearRelative
 
 
 def test_prior():
-    pose = Prior(Z=Normal(0, 2))
+    pose = Prior(Normal(0, 2))
```

Comment:
yeah, Julia's `Base.@kwdef` is pretty cool.  The original "direct" constructors are still there, but the full keyword constructor with optional defaults and required fields are constructed for you at compile time.

Reply:
Er, this is Python :D The equivalent Julia version is a dataclass in Python, it's got a few more bells and whistles but similar concept.

Reply:
I'm going to take this as a compliment that the API's look almost exactly the same :tada: